yt_dlp/extractor/fancode.py

```diff
@@ -63,22 +66,24 @@ def _real_extract(self, url):
             }'''
 
         metadata_json = self._download_json(GQL_URL, video_id, data=str.encode(data % video_id), headers=headers, note="Downloading metadata")
-        brightcove_video_id = metadata_json.get('data').get('media').get('mediaSource').get('brightcove')
+        brightcove_video_id = try_get(metadata_json, lambda x: x['data']['media']['mediaSource']['brightcove'], compat_str)
 
         if brightcove_video_id is None:
             raise RegexNotFoundError('Unable to extract brightcove Video ID')
 
-        if metadata_json.get('data').get('media').get('isPremium') is True:
+        if try_get(metadata_json, lambda x: x['data']['media']['isPremium']) is True:
             self.report_warning("requires a premium account" % video_id, video_id)
 
+        media = try_get(metadata_json, lambda x: x['data']['media'])
```

Comment:
You can move `media` above the premium check, no?
then you can do `media.get('isPremium')`